data-protocols/dsp/dsp-api-configuration/src/main/java/org/eclipse/edc/protocol/dsp/api/configuration/auth/ClaimTokenRequestFilter.java

```diff
@@ -0,0 +1,98 @@
+/*
+ *  Copyright (c) 2023 Fraunhofer Institute for Software and Systems Engineering
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Fraunhofer Institute for Software and Systems Engineering - initial API and implementation
+ *
+ */
+
+package org.eclipse.edc.protocol.dsp.api.configuration.auth;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.ws.rs.container.ContainerRequestContext;
+import jakarta.ws.rs.container.ContainerRequestFilter;
+import org.eclipse.edc.spi.EdcException;
+import org.eclipse.edc.spi.iam.ClaimToken;
+import org.eclipse.edc.spi.iam.IdentityService;
+import org.eclipse.edc.spi.iam.TokenRepresentation;
+import org.eclipse.edc.spi.result.Result;
+import org.eclipse.edc.web.spi.exception.AuthenticationFailedException;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * Intercepts all incoming requests and authenticates them using the IdentityService. As a ClaimToken
+ * instance is required for further processing of the claims, the ClaimToken returned by the
+ * IdentityService is serialized and added as an additional header.
+ */
+public class ClaimTokenRequestFilter implements ContainerRequestFilter {
```

Comment:
I would also change the approach, personally I'm more for the number 2, it will be just simpler to call the `IdentityService` directly in the controller, throwing the authentication failed exception when the result is failed

Reply:
Removed the request filter. Authentication will then be handled in the controllers.